docs/DeveloperGuide.md

```diff
@@ -1,29 +1,159 @@
+
+`ExitCommand`: Deliver a `CommandResult` to terminate software run.
+
+`InvalidCommand`: Deliver a `CommandResult` of invalid command
+
+`ListCommand`: Print all formatted `Recipe` in `RecipeList` to screen
+
+`ViewCommand`: View an existing `Recipe` at a valid index from `RecipeList`,
+otherwise show error message on `index out of bound`
+
+### Recipe Module
+The recipe module encapsulates the array, recipe and ingredient objects.
+
+<p align="center" width="100%">
+  <img width="80%" src="images/RecipeModuleDiagram.png" alt="Recipe Module Diagram"/>
+</p>
+
+**API:** `RecipeList.java`
+1. `RecipeList` calls `Recipe` to add new recipe to the list
+2. `RecipeList` calls `Recipe` to edit existing recipes
+3. `RecipeList` calls `Recipe` to delete a recipe
+
+**API:** `Recipe.java`
+1. `Recipe` calls `Ingredient` to add, edit or delete ingredients
 
 ## Product scope
 ### Target user profile
 
-{Describe the target user profile}
+Target user of the application is avid cook who wants to organize their recipe list for ease of reference and search 
 
 ### Value proposition
 
 {Describe the value proposition: what problem does it solve?}
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+|Version| As a ... | I want to ...                                 | So that I can ...                                                                               |
+|--------|---------|-----------------------------------------------|-------------------------------------------------------------------------------------------------|
+|v1.0|new user| see usage instructions                        | refer to them when I forget how to use the application                                          |
+|v1.0|user| add new recipes                               | store recipes that I need without having to remember everything                                 |
+|v1.0|user| delete existing recipes                       | remove recipes that I no longer use so that the list will not be cluttered                      |
+|v1.0|user| show all recipes in my list                   | view an overview of what recipes I have added beforehand                                        |
+|v1.0|user| show detailed recipe that I specified         | view detailed recipe (name, description, ingredients and steps) of the one that I am interested |
+|v1.0|user| exit the application                          | close the entire application                                                                    |
+|v2.0|user| find recipe by recipe name or ingredient name | locate a recipe without having to go through the entire list    g                               |
 
 ## Non-Functional Requirements
 
```

Comment:
Most technical detail suggestions are given by other reviewers. For layout-wise, maybe you could consider making the DG more "layered", ie. the reviewer has a sense of which part/chapter they are reading now. Currently, the DG looks "flat", where every piece of information looks like they have the same importance. Even though markdown has limitations in formatting, you could consider using larger / smaller fonts to tidy up the DG more.